File: C/Palindromo.c
/*Faça um programa que e recebe uma string como parametro e
retorna true se essa é um palindromo. Na saıda padrão, para cada linha de entrada, escreva
uma linha de saída com SIM/NAO indicando se a linha é um palíndromo. Destaca-se que uma 
linha de entrada pode ter caracteres não letras. A entrada termina com a leitura de FIM.*/
//Autor: Henrique Grissi
#include <string.h>
#include <stdio.h>
#include<stdbool.h>
#define TAM 700
int fim (char r[TAM],int tam){
    int resp = 0; //False
    if(tam == 2 && r[0] == 'F' && r[1] == 'I' && r[2] == 'M'){
        return resp;
    }else{
        resp = 1;//true
    }
    return resp;
}
int main (void){
    char palavra[700];
    int tam, c; //Váriável do tamanho da string
    fgets(palavra, 700, stdin);
    tam = strlen(palavra);//Usando a função "strlen" para saber qual é o tamanho da string (não considera o caractere NULL)
    tam -= 2; //Ajustando para o tamanho do array
    c = fim(palavra,tam);
    while(c){
        for(int i=0;i<=tam;i++){
            if(palavra[i] != palavra[tam]){
              printf("NAO\n");
              break; //Interrompe o loop, para não printar "NÃO" mais de uma vez 
            }
            else{
                printf("SIM\n");
                break; //Interrompe o loop, para não printar "SIM" mais de uma vez 
            }
            tam--;
        }
      //Recebendo informações da nova palavra que será lida
      fgets(palavra, 700, stdin); //Input
      tam = strlen(palavra); //Saber o tamanho da nova String 
      tam -= 2; //Ajustando o tamanho para o tamanho do array.
      c = fim(palavra,tam); //Chama a função para testar "FIM"
    }
  return 0;
}

File: JAVA/Palindromo.java
/*Crie um m ́etodo iterativo que recebe uma string como parˆametro e
retorna true se essa  ́e um pal ́ındromo. Na sa ́ıda padr ̃ao, para cada linha de entrada, escreva
uma linha de sa ́ıda com SIM/N  ̃AO indicando se a linha  ́e um pal ́ındromo. Destaca-se que uma
linha de entrada pode ter caracteres n ̃ao letras. A entrada termina com a leitura de FIM.*/

import java.util.Scanner;  
public class Palindromo {
    public static boolean fim (String r){
        boolean fim = true;
        if (r.charAt(0) == 'F' && r.charAt(1) == 'I' && r.charAt(2) == 'M'){
            fim = false;
            return fim;
        }else{
            return fim;
        }

    }
    public static boolean palindromo (String p){
        int tam = 0;
        boolean pali = false;
        tam = p.length(); //atribuindo a variável tam o TAMANHO da string 
        tam--;
        for (int i=0; i<=tam; i++){
            if(p.charAt(i) != p.charAt(tam)){
                return pali;
            }else{
                pali = true;
            }
            tam--;
        }
        return pali;
    }
    public static void main(String args[]) {
        Scanner leitor = new Scanner(System.in);
        boolean resp = false; //True = palindromo, false = não palindromo
        String palavra = MyIO.readLine();
        while (fim(palavra)){  //Chama o método para confirmar se a palavra é "FIM" (enquanto for true o loop continua)
            resp = palindromo(palavra); //Chama o método de verificação do palíndromo
            if(resp==true){
                System.out.println("SIM");
            }else{
                System.out.println("NAO");
            }
            palavra = MyIO.readLine();
        }
        leitor.close();//"Fechando" o scanner 
    }         
}
